mypy/checker.py

```diff
@@ -2188,7 +2188,14 @@ def check_method_override_for_base_with_name(
         else:
             override_class_or_static = defn.func.is_class or defn.func.is_static
         typ, _ = self.node_type_from_base(defn.name, defn.info, defn)
-        assert typ is not None
+        if typ is None:
+            # This may only happen if we're checking `x-redefinition` member
+            # and `x` itself is for some reason gone. Normally the node should
+            # be reachable from the containing class by its name.
+            # The redefinition is never removed, use this as a sanity check to verify
+            # the reasoning above.
+            assert f"{defn.name}-redefinition" in defn.info.names
```

Comment:
@sterliakov What about multiple redefinitions? Will this fix still work? The current logic in `semanal.py` is
```python
            if i == 1:
                new_name = f"{name}-redefinition"
            else:
                new_name = f"{name}-redefinition{i}"
```

Reply:
The first redefinition cannot be `InitVar`, so should not get deleted. I can replace this with `any(name.startswith(f"{defn.name}-redifinition") for name in info.names)`, but I couldn't invent the case where `name-redefinition` is also gone

Reply:
OK, if you can't trigger it, it is fine.